### PR TITLE
Downscale Images to document resolution

### LINKED ISSUE #1046: Add a downscale control to the Image node for performance

### PR DESCRIPTION
**Pull request by TrueDoctor**
Closes #1046

### CODE REVIEW

File: node-graph/graph-craft/src/document.rs [hunk 1]
```diff
@@ -127,6 +128,11 @@ pub enum NodeInput {
 	Node { node_id: NodeId, output_index: usize, lambda: bool },
 	Value { tagged_value: crate::document::value::TaggedValue, exposed: bool },
 	Network(Type),
+	// A Lambda input represents an input that is not resolved through function composition but
+	// actually consuming the provided input instead of passing it to its predecessor
+	// TODO: Find a better name to avoid confusion with node parameters as lambdas, e.g. CallArg,
+	// FunctionArgument, EvalInput, ConsumeInput
+	Lambda(Type),
```
Comment (Keavon) [inline node-graph/graph-craft/src/document.rs:135]: Let's talk about this and pick a good name together before merging this.

File: node-graph/gcore/src/raster.rs [hunk 2]
```diff
@@ -329,16 +329,27 @@ mod image {
...
 		pub height: u32,
 		pub data: Vec<Color>,
 	}
 
+	impl Hash for Image {
+		fn hash<H: Hasher>(&self, state: &mut H) {
+			const HASH_SAMPLES: usize = 1000;
+			self.width.hash(state);
+			self.height.hash(state);
+			for i in 0..HASH_SAMPLES.min(self.data.len()) {
```
Comment (Keavon) [inline node-graph/gcore/src/raster.rs:347]: We only hash based on the first thousand pixels? Seems pretty error-prone.

File: Cargo.toml [hunk 3]
```diff
@@ -24,19 +24,26 @@ members = [
...
+
+[profile.dev.package.graphene-core]
+opt-level = 1
+
+[profile.dev.package.graphene-std]
+opt-level = 1
 
 [profile.dev.package.graphite-wasm]
-opt-level = 3
+opt-level = 1
```
Comment (Keavon) [inline Cargo.toml:46]: Shouldn't this work by optimizing all deps fully, except our own first-party workspace crates? I see how this explicitly optimizes autoquant, image, etc. but not everything third-party.

**Review by Keavon — APPROVED** [review]

File: node-graph/gcore/src/raster.rs (hunk 2 above)
Comment (TrueDoctor) [reply node-graph/gcore/src/raster.rs:347]: We hash based on 1000 pixels evenly spaced on the image

File: node-graph/graph-craft/src/document.rs (hunk 1 above)
Comment (TrueDoctor) [reply node-graph/graph-craft/src/document.rs:135]: Discussion: `NodeInput::ShortCircut`

**Review by Keavon — APPROVED** [review]

**Pull request merged**